src/shogun/mathematics/linalg/internal/implementation/Scale.h

```diff
@@ -66,6 +66,34 @@ struct scale
 	static void compute(Matrix A, Matrix B, Matrix C, T alpha, T beta);
 };
 
+/** Specialization of scale for the Native backend */
+template<class Matrix>
+struct scale<Backend::NATIVE, Matrix>
+{
+	typedef typename Matrix::Scalar T;
+	
+	/** Performs the operation B = alpha*A */
+	static void compute(SGMatrix<T> A, SGMatrix<T> B, T alpha)
+	{
+		REQUIRE((A.num_rows==B.num_rows)&&(A.num_cols==B.num_cols),
+			"Dimensions of A(%dx%d) and B(%dx%d) don't match.\n",
+			A.num_rows, A.num_cols, B.num_rows, B.num_cols);
+		
+		for (int32_t i=0; i<A.num_rows*A.num_cols; i++)
+			B[i]=A[i]*alpha;
+	}
+	
+	/** Performs the operation B = alpha*A */
+	static void compute(SGVector<T> A, SGVector<T> B, T alpha)
+	{
+		REQUIRE(A.vlen==B.vlen,"Number of elements in A(%d) should be "
+			"equal to number of elements in (%d)!\n", A.vlen, B.vlen);
+		
+		for (int32_t i=0; i<A.vlen; i++)
+			B[i]=A[i]*alpha;
+	}
+};
+
```

Comment:
Yeah. Keep the other requires there - they are useful. The null check REQUIRE is useful when people (hopefully mistakenly) passes scale(NULL, 1000, 0.5) in the raw ptr method.